source/enum_set.h

```diff
@@ -224,56 +238,69 @@ class EnumSet {
       return 0;
     }
 
-    size_t index =
-        std::min(buckets_.size() - 1, ComputeTheoreticalBucketIndex(value));
-    const T needle = ComputeBucketStart(value);
+    size_t index = std::min(buckets_.size() - 1,
+                            ComputeLargestPossibleBucketIndexFor(value));
```

Comment:
I have two suggestion that could remove a lot of the special cases.

```suggestion
    size_t index = std::min(buckets_.size(),
                            ComputeLargestPossibleBucketIndexFor(value)+1);
```

With this I would remove lines 245 to 259 because they are unnecessary.

Reply:
Good point. I kept the -1 still to get rid of the pre index--, which seemed weird.
Added some comments etc, but taking this, way shorter code.